styx-service-common/src/main/java/com/spotify/styx/storage/InstrumentedBatch.java

```diff
@@ -21,12 +21,44 @@
 package com.spotify.styx.storage;
 
 import com.google.cloud.datastore.Batch;
+import com.google.cloud.datastore.Datastore;
+import com.google.cloud.datastore.DatastoreBatchWriter;
 import com.google.cloud.datastore.DatastoreWriter;
+import com.google.cloud.datastore.Entity;
+import com.google.cloud.datastore.FullEntity;
 import com.spotify.styx.monitoring.Stats;
+import java.util.List;
 import java.util.Objects;
 
 interface InstrumentedBatch extends Batch, InstrumentedDatastoreBatchWriter {
```

Comment:
Took me awhile to understand the relationship between those interfaces. Would it be simpler making this a class instead of interface?

Reply:
So something like
```
class InstrumentedBatch implements Batch, InstrumentedDatastoreBatchWriter {

  private final Stats stats;
  private final Batch batch;

  InstrumentedBatch(Stats stats, Batch batch) {
    this.stats = Objects.requireNonNull(stats, "stats");
    this.batch = Objects.requireNonNull(batch, "batch");
  }

  @Override
  public Entity add(FullEntity<?> entity) {
    return InstrumentedDatastoreBatchWriter.super.add(entity);
  }

  @Override
  public List<Entity> add(FullEntity<?>... entities) {
    return InstrumentedDatastoreBatchWriter.super.add(entities);
  }

  @Override
  public Response submit() {
    return batch.submit();
  }

  @Override
  public Datastore getDatastore() {
    return InstrumentedDatastore.of(batch.getDatastore(), stats());
  }

  @Override
  public Stats stats() {
    return stats;
  }

  @Override
  public DatastoreBatchWriter batchWriter() {
    return batch;
  }

  @Override
  public DatastoreWriter writer() {
    return batch;
  }
}
```

Reply:
I'd like to keep refactoring out of the scope of this PR.

Reply:
Ok fair enough.